smokefree/sites/all/modules/custom/icfi_api_keys/ICFIApi.inc

```diff
@@ -0,0 +1,104 @@
+<?php
+
+
+class ICFIApi {
+  protected $machineName;
+  protected $apiValue;
+  protected $isJsVariable;
+
+  public function __construct($machineName, $apiValue, $isJsVariable) {
+    if (empty($machineName) || empty($apiValue)) {
+      throw new InvalidArgumentException('Please supply a valid api value and an appropriate machine name');
+    }
+
+    // Remove accidental whitespaces at the end
+    $this->machineName = trim($machineName);
+    $this->apiValue = trim($apiValue);
+    $this->isJsVariable = $isJsVariable;
+  }
+
+  /**
+   * Create a new API key entry in the database
+   */
+  public function create() {
+
+    // Check if this machine name exists, only enter a new record if there is no existing value
+    $entryExists = self::getKeyValue($this->machineName);
+    if (empty($entryExists)) {
+      $transaction = db_transaction();
+      try {
+        $query = db_insert('icfi_api_keys')->fields(array(
+            'api_key_machine_name' => $this->machineName,
+            'api_key_value' => $this->apiValue,
+            'javascript_variable' => $this->isJsVariable,)
+        );
+        $query->execute();
+        return TRUE;
+      }
+      //Shouldn't really happen
+      catch (Exception $e) {
+        $transaction->rollback();
+        watchdog_exception('icfi_api_keys', $e);
+        return FALSE;
```

Comment:
Should this return false or throw exception. If it is returning false then there is no way of distinguishing whether create method is returning false because you have chosen a key that already exists vs. because there was an issue with the database insertion, but these two scenarios require different responses.

Reply:
Yea, that makes sense. With the exception is being correctly handled in the unlikely event of a database error/failure. I'll remove that.

Returning "FALSE" will be an indicator of choosing a key that already exists